lib/Dialect/TCP/IR/TCPOps.cpp

```diff
@@ -9,11 +9,81 @@
 #include "npcomp/Dialect/TCP/IR/TCPOps.h"
 #include "mlir/Dialect/Shape/IR/Shape.h"
 #include "mlir/IR/TypeUtilities.h"
+#include "llvm/ADT/STLExtras.h"
 
 using namespace mlir;
 using namespace mlir::NPCOMP;
 using namespace mlir::NPCOMP::tcp;
 
+//===----------------------------------------------------------------------===//
+// TensorToMemrefOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult TensorToMemrefOp::fold(ArrayRef<Attribute> operands) {
+  if (auto memrefToTensor = tensor().getDefiningOp<tcp::MemrefToTensorOp>())
+    return memrefToTensor.memref();
+  return nullptr;
+}
+
+//===----------------------------------------------------------------------===//
+// ShapedResultsOp
+//===----------------------------------------------------------------------===//
+
+void ShapedResultsOp::build(OpBuilder &builder, OperationState &result,
+                            TypeRange resultTypes, ValueRange resultShapes) {
+  result.addOperands(resultShapes);
+  result.addTypes(resultTypes);
+  (void)result.addRegion();
```

Comment:
Why do you need a (void) here?

Reply:
It returns a Region, but we don't use it. Not strictly needed, but saw this convention somewhere else. (maybe it appeases some warning on some compiler?)